tests/ui/traits/new-solver/unsize-although-ambiguous.rs

```diff
@@ -0,0 +1,13 @@
+// check-pass
+// compile-flags: -Ztrait-solver=next
+
+use std::fmt::Display;
+
+fn box_dyn_display(_: Box<dyn Display>) {}
+
+fn main() {
+    // During coercion, we don't necessarily know whether `{integer}` implements
+    // `Display`. Before, that would cause us to bail out in the coercion loop when
+    // checking `{integer}: Unsize<dyn Display>`.
```

Comment:
I can rewrite this to be more detailed if you'd like. 

The gist is that we should be able to select `{integer}: Unsize<dyn Display>` successfully, resulting in an `ImplSource::Builtin` with two nested goals -- `{integer}: Display` and `{integer}: Sized`. Even though the first goal is ambiguous, we're ok with that during coercion, since we only ever call loop on on `Unsize` and `CoerceUnsized` goals, and just stash the rest for later:
https://github.com/rust-lang/rust/blob/9227ff28aff55b252314076fcf21c9a66f10ac1e/compiler/rustc_hir_typeck/src/coercion.rs#L654-L657

Reply:
can you write a test where this unsize behavior results in us incorrectly erroring?

Reply:
I don't think such a test can be constructed, actually.